File: app/src/main/java/ollu/dp/ua/weather/event_bus/Params.java
package ollu.dp.ua.weather.event_bus;

import android.support.annotation.NonNull;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.Date;

/**
 * Класс на основе JSONObject, дополненный возможностью хранить объект типа Date.
 */

@SuppressWarnings("unused")
public class Params extends JSONObject {

    @Override
    public JSONObject put(@NonNull String name, Object value) {
        try {
            super.put(name, value);
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return this;
    }

    @Override
    public JSONObject put(@NonNull String name, int value) {
        return put(name, (Object)value);
    }

    @Override
    public JSONObject put(@NonNull String name, long value) {
        return put(name, (Object)value);
    }

    @Override
    public JSONObject put(@NonNull String name, double value) {
        return put(name, (Object)value);
    }

    @Override
    public JSONObject put(@NonNull String name, boolean value) {
        return put(name, (Object)value);
    }

    public JSONObject put(@NonNull String name, Date value) {
        return put(name, Long.valueOf(value.getTime()));
    }

    @SuppressWarnings("SameParameterValue")
    private String getString(@NonNull String name, String defaultValue) {
        try {
            return super.getString(name);
        } catch (JSONException e) {
            e.printStackTrace();
            return defaultValue;
        }
    }

    @Override
    public String getString(@NonNull String name) {
        return getString(name, "");
    }

    public long getLong(@NonNull String name, long defaultValue) {
        try {
            return super.getLong(name);
        } catch (JSONException e) {
            e.printStackTrace();
            return defaultValue;
        }
    }

    public int getInt(@NonNull String name, int defaultValue) {
        try {
            return super.getInt(name);
        } catch (JSONException e) {
            e.printStackTrace();
            return defaultValue;
        }
    }

    public boolean getBoolean(@NonNull String name, boolean defaultValue) {
        try {
            return super.getBoolean(name);
        } catch (JSONException e) {
            e.printStackTrace();
            return defaultValue;
        }
    }

    public double getDouble(@NonNull String name, double defaultValue) {
        try {
            return super.getDouble(name);
        } catch (JSONException e) {
            e.printStackTrace();
            return defaultValue;
        }
    }

    public Date getDate(@NonNull String name, Date defaultValue) {
        try {
            return new Date(getLong(name));
        } catch (Exception e) {
            e.printStackTrace();
            return defaultValue;
        }
    }


}

File: app/src/main/java/ollu/dp/ua/weather/event_bus/Params.kt
@file:Suppress("PackageName")

package ollu.dp.ua.weather.event_bus

import org.json.JSONException
import org.json.JSONObject
import java.util.*

/**
 * Класс на основе JSONObject, дополненный возможностью хранить объект типа Date.
 */

@Suppress("unused")
open class Params : JSONObject() {

    override fun put(name: String, value: Any): JSONObject {
        try {
            super.put(name, value)
        } catch (e: JSONException) {
            e.printStackTrace()
        }

        return this
    }

    override fun put(name: String, value: Int): JSONObject {
        return put(name, value as Any)
    }

    override fun put(name: String, value: Long): JSONObject {
        return put(name, value as Any)
    }

    override fun put(name: String, value: Double): JSONObject {
        return put(name, value as Any)
    }

    override fun put(name: String, value: Boolean): JSONObject {
        return put(name, value as Any)
    }

    fun put(name: String, value: Date): JSONObject {
        return put(name, java.lang.Long.valueOf(value.time))
    }

    private fun getString(name: String, defaultValue: String): String {
        return try {
            super.getString(name)
        } catch (e: JSONException) {
            e.printStackTrace()
            defaultValue
        }

    }

    override fun getString(name: String): String {
        return getString(name, "")
    }

    fun getLong(name: String, defaultValue: Long): Long {
        return try {
            super.getLong(name)
        } catch (e: JSONException) {
            e.printStackTrace()
            defaultValue
        }

    }

    fun getInt(name: String, defaultValue: Int): Int {
        return try {
            super.getInt(name)
        } catch (e: JSONException) {
            e.printStackTrace()
            defaultValue
        }

    }

    fun getBoolean(name: String, defaultValue: Boolean): Boolean {
        return try {
            super.getBoolean(name)
        } catch (e: JSONException) {
            e.printStackTrace()
            defaultValue
        }

    }

    fun getDouble(name: String, defaultValue: Double): Double {
        return try {
            super.getDouble(name)
        } catch (e: JSONException) {
            e.printStackTrace()
            defaultValue
        }

    }

    fun getDate(name: String, defaultValue: Date): Date {
        return try {
            Date(getLong(name))
        } catch (e: Exception) {
            e.printStackTrace()
            defaultValue
        }

    }


}
